examples/notebook_tutorials/random_walk_tutorial.ipynb

```diff
@@ -0,0 +1,960 @@
+    "classical one, showing that the process of a QW is quadratically faster than its classical counterpart.\n",
+    "\n",
+    "\n",
+    "In order to create a quantum walk, we have to translate the components of the classical random walk \n",
+    "to a quantum problem. We can encode the position of a \"walker\" in some $n$ -dimensional space with a vector\n",
+    "$\\lvert j\\rangle$. For the purpose of this project, we will be investigating a basic case of a \n",
+    "random walk on a ring-shaped graph, with adjacent nodes connected by a single edge. The configuration \n",
+    "looks something like this:\n",
+    "\n",
+    "\n",
+    "<img src=\"assets/cycle.png\" style=\"width:550px;height:500px;\">\n",
+    "\n",
+    "\n",
+    "Going back to our original idea of some position vector $\\lvert j\\rangle$, it is apparent that in order to \n",
+    "encode the location of a \"walker\" on this graph, we need to assign a specific value of our position vector \n",
+    "to each node. Well, this is fairly simple, for a graph of $K$ nodes, we form a Hilbert space\n",
+    "$H_W$ spanned by the following set:\n",
+    "\n",
+    "\n",
+    "$$H_W \\ = \\ \\{\\lvert j\\rangle \\ : \\ j \\ = \\ 0, \\ ..., \\ K \\ - \\ 1 \\}$$\n",
+    "\n",
+    "\n",
+    "We also require another vector in order to create a random walk. We need a \"coin vector\", which will encode \n",
+    "the direction in which the random walk will progress at the $T$-th step of the process. This Hilbert space \n",
+    "is spanned by the two basis states, representing forward and backward progression on our number-line-like \n",
+    "graph (actually, I guess our graph looks more like a ring, so I guess our two basis states will represent \n",
+    "clockwise and counter-clockwise motion, but it's the same idea). We will call this Hilbert space $H_C$, \n",
+    "and we can again define our spanning set:\n",
+    "\n",
+    "\n",
+    "$$H_C \\ = \\ \\{\\lvert i\\rangle \\ : \\ i \\ = \\ \\downarrow, \\ \\uparrow\\rangle\\}$$\n",
+    "\n",
+    "\n",
+    "Where the upward-arrow symbol represent counter-clockwise motion, and the downward arrow represents \n",
+    "clock-wise motion. Now that we have defined all of the vectors we need to encode the information about \n",
+    "our random walk, we must understand how we can realize these vectors in our quantum algorithm. Well, this is \n",
+    "again fairly simple. For a graph of $K \\ = \\ 2^n$ nodes, we require $n$ qubits to encode binary representations \n",
+    "of numbers ranging from $0$ to $K \\ - \\ 1$, therefore each of the vectors spanning $H_W$ will be given \n",
+    "by the binary representation of $j$ corresponding to the basis vector $\\lvert j\\rangle$. For the coin vector, \n",
+    "since we have only two states, we only need one qubit to encode the two possible states:\n",
+    "\n",
+    "\n",
+    "$$\\lvert 0\\rangle \\ = \\ \\lvert \\uparrow\\rangle \\ \\ \\text{and} \\ \\ 
```

Comment:
Where do you actually use this function in this tutorial? If you don't it should be removed